dev/src/expression.ts

```diff
@@ -0,0 +1,6739 @@
+   * firestore.pipeline().collection("items")
+   *   .addFields(Field.of("price").multiply(Field.of("quantity")).as("totalPrice"));
+   * ```
+   *
+   * @param name The alias to assign to this expression.
+   * @return A new {@link ExprWithAlias} that wraps this
+   *     expression and associates it with the provided alias.
+   */
+  as(name: string): ExprWithAlias<typeof this> {
+    return new ExprWithAlias(this, name);
+  }
+
+  abstract _toProto(serializer: Serializer): api.IValue;
+}
+
+/**
+ * @beta
+ */
+export class ExprWithAlias<T extends Expr> extends Expr implements Selectable {
+  exprType: ExprType = 'ExprWithAlias';
+  selectable = true as const;
+
+  constructor(
+    public expr: T,
+    public alias: string
+  ) {
+    super();
+  }
+
+  _toProto(serializer: Serializer): api.IValue {
+    throw new Error('ExprWithAlias should not be serialized directly.');
+  }
+}
+
+/**
+ * @internal
+ */
+class ListOfExprs extends Expr {
+  exprType: ExprType = 'ListOfExprs';
+  constructor(private exprs: Expr[]) {
+    super();
+  }
+
+  _toProto(serializer: Serializer): api.IValue {
+    return {
+      arrayValue: {
+        values: this.exprs.map(p => serializer.encodeValue(p)!),
+      },
+    };
+  }
+}
+
+/**
+ * @beta
+ *
+ * Represents a reference to a field in a Firestore document, or outputs of a {@link Pipeline} stage.
+ *
+ * <p>Field references are used to access document field values in expressions and to specify fields
+ * for sorting, filtering, and projecting data in Firestore pipelines.
+ *
+ * <p>You can create a `Field` instance using the static {@link #of} method:
+ *
+ * ```typescript
+ * // Create a Field instance for the 'name' field
+ * const nameField = Field.of("name");
+ *
+ * // Create a Field instance for a nested field 'address.city'
+ * const cityField = Field.of("address.city");
+ * ```
+ */
+export class Field extends Expr implements Selectable {
+  exprType: ExprType = 'Field';
+  selectable = true as const;
+
+  private constructor(
+    private fieldPath: firestore.FieldPath,
+    private pipeline: Pipeline | null = null
+  ) {
+    super();
+  }
+
+  /**
+   * Creates a {@code Field} instance representing the field at the given path.
+   *
+   * The path can be a simple field name (e.g., "name") or a dot-separated path to a nested field
+   * (e.g., "address.city").
+   *
+   * ```typescript
+   * // Create a Field instance for the 'title' field
+   * const titleField = Field.of("title");
+   *
+   * // Create a Field instance for a nested field 'author.firstName'
+   * const authorFirstNameField = Field.of("author.firstName");
+   * ```
+   *
+   * @param name The path to the field.
+   * @return A new {@code Field} instance representing the specified field.
+   */
+  static of(name: string): Field;
+  static of(path: firestore.FieldPath): Field;
+  static of(nameOrPath: string | firestore.FieldPath): Field;
```

Comment:
This overload is redundant. `static of(nameOrPath: string | firestore.FieldPath): Field;` Either remove this or keep it and remove the two above.